BDArmory/Weapons/Missiles/MissileLauncher.cs

```diff
@@ -2831,5 +2902,23 @@ void DetachExhaustPrefabs()
             exhaustPrefabs.Clear();
         }
         #endregion
+        public double GetDeltaV()
+        {
+            double specificImpulse;
+            double deltaV;
+            double massFlowRate;
+
+            massFlowRate = (boostTime == 0) ? 0 : boosterMass / boostTime;
+            specificImpulse = (massFlowRate == 0) ? 0 : thrust / (massFlowRate * 9.81);
+            deltaV = specificImpulse * 9.81 * Math.Log(part.mass / (part.mass - boosterMass));
+
+            double mass = part.mass;
+            massFlowRate = (cruiseTime == 0) ? 0 : sustainerMass / cruiseTime;
+            if (boosterMass > 0) mass -= boosterMass;
+            specificImpulse = (massFlowRate == 0) ? 0 : cruiseThrust / (massFlowRate * 9.81);
+            deltaV += (specificImpulse * 9.81 * Math.Log(mass / (mass - sustainerMass)));
+
+            return deltaV;
+        }
     }
```

Comment:
Having read though the fuelburn stuff, it might make more sense to have either A): boost/cruse time be dynamically determined by fuel mass, (either via a hardcoded Isp (270?) or have it be a settable value somewhere if it absolutely needs to be adjustable), or B): fuelmass be determined by thrust + thrustTime, since else there's nothing to prevent someone from setting thrust to e.g. 100 and fuelmass to 1kg, and enjoying the resulting multi-thousand ISP Torchdrive-powered missiles.
Getting Missile dV also seems most relevant if in space, as in atmo the average user seeing missile X has Y dV isn't going to mean much, compared to missile has X range and Y speed

Reply:
I've thought about the code limitation since I already implement an ISP calculation, but I've come to the conclusion that it depends on each configuration. Even without FuelBurn, someone can still set up a configuration with a small liftArea and a small weight, and crank up the thrust and boostTime, resulting in an absurd ISP. So, I thought about leaving it up to each individual configuration, maybe generating a default configuration based on thrust + thrustTime, but allowing it to be altered through the config, like a FloatCurve.

Reply:
i created this DefaultMass() to set the booster/sustainer mass to be 10% of the total mass each if the useFuel is true and it don't have a default value